\documentclass{beamer}\usepackage[]{graphicx}\usepackage[]{color}
%% maxwidth is the original width if it is less than linewidth
%% otherwise use linewidth (to make sure the graphics do not exceed the margin)
\makeatletter
\def\maxwidth{ %
  \ifdim\Gin@nat@width>\linewidth
    \linewidth
  \else
    \Gin@nat@width
  \fi
}
\makeatother

\definecolor{fgcolor}{rgb}{0.345, 0.345, 0.345}
\newcommand{\hlnum}[1]{\textcolor[rgb]{0.686,0.059,0.569}{#1}}%
\newcommand{\hlstr}[1]{\textcolor[rgb]{0.192,0.494,0.8}{#1}}%
\newcommand{\hlcom}[1]{\textcolor[rgb]{0.678,0.584,0.686}{\textit{#1}}}%
\newcommand{\hlopt}[1]{\textcolor[rgb]{0,0,0}{#1}}%
\newcommand{\hlstd}[1]{\textcolor[rgb]{0.345,0.345,0.345}{#1}}%
\newcommand{\hlkwa}[1]{\textcolor[rgb]{0.161,0.373,0.58}{\textbf{#1}}}%
\newcommand{\hlkwb}[1]{\textcolor[rgb]{0.69,0.353,0.396}{#1}}%
\newcommand{\hlkwc}[1]{\textcolor[rgb]{0.333,0.667,0.333}{#1}}%
\newcommand{\hlkwd}[1]{\textcolor[rgb]{0.737,0.353,0.396}{\textbf{#1}}}%

\usepackage{framed}
\makeatletter
\newenvironment{kframe}{%
 \def\at@end@of@kframe{}%
 \ifinner\ifhmode%
  \def\at@end@of@kframe{\end{minipage}}%
  \begin{minipage}{\columnwidth}%
 \fi\fi%
 \def\FrameCommand##1{\hskip\@totalleftmargin \hskip-\fboxsep
 \colorbox{shadecolor}{##1}\hskip-\fboxsep
     % There is no \\@totalrightmargin, so:
     \hskip-\linewidth \hskip-\@totalleftmargin \hskip\columnwidth}%
 \MakeFramed {\advance\hsize-\width
   \@totalleftmargin\z@ \linewidth\hsize
   \@setminipage}}%
 {\par\unskip\endMakeFramed%
 \at@end@of@kframe}
\makeatother

\definecolor{shadecolor}{rgb}{.97, .97, .97}
\definecolor{messagecolor}{rgb}{0, 0, 0}
\definecolor{warningcolor}{rgb}{1, 0, 1}
\definecolor{errorcolor}{rgb}{1, 0, 0}
\newenvironment{knitrout}{}{} % an empty environment to be redefined in TeX

\usepackage{alltt}
\usetheme{Boadilla}
\usecolortheme{beaver}


\usepackage{graphicx}
\graphicspath{{../figures/}}
\usepackage{amsmath}
\usepackage{booktabs}

\author{Christof Angermueller}
\title{Statistics Introduction}
\date{\today}





\IfFileExists{upquote.sty}{\usepackage{upquote}}{}
\begin{document}

\begin{frame}
  \titlepage
\end{frame}

\begin{frame}
  \tableofcontents
\end{frame}



\section{Clustering}
\begin{frame}
\tableofcontents[currentsection]
\end{frame}

\begin{frame}[fragile]{Clustering}
  \begin{columns}
    \begin{column}{.5\linewidth}
      \begin{itemize}
        \item \textbf{Goal}: Finding groups of related items
          \vspace{1cm}
        \item How do define relatedness?
        \item Which clustering methods exist?
        \item What is Hierarchical clustering?
        \item How to visualize clustering?
      \end{itemize}
    \end{column}
    \begin{column}{.5\linewidth}
      \begin{center}
        \includegraphics[width=.8\linewidth]{clust.png}
      \end{center}
    \end{column}
  \end{columns}
\end{frame}

\begin{frame}[fragile]{Methods}
  \begin{columns}[t]
    \begin{column}{.32\linewidth}
      \begin{block}{Partitioning clustering}
        \begin{itemize}
          \item Partition points into k clusters
          \item k known a-priori
          \item k-means
          \item k-medoids
        \end{itemize}
        \begin{center}
          \includegraphics[width=\linewidth]{clust_part.png}
        \end{center}
      \end{block}
    \end{column}
    \begin{column}{.32\linewidth}
      \begin{block}{Density clustering}
        \begin{itemize}
          \item Cluster points into dense regions
          \item k unknown a-priori
          \item DBSCAN
          \item OPTICS
        \end{itemize}
        \begin{center}
          \includegraphics[width=\linewidth]{clust_dens.png}
        \end{center}
      \end{block}
    \end{column}
    \begin{column}{.32\linewidth}
      \begin{block}{Hierarchical clustering}
        \begin{itemize}
          \item Find hierarchy of clusters
          \item k unknown a-priori
          \item Single-linkage
          \item Complete-linkage
          \item Average-linkage
        \end{itemize}
        \begin{center}
          \includegraphics[width=.6\linewidth]{clust_hier.png}
        \end{center}
      \end{block}
    \end{column}
  \end{columns}
\end{frame}

\begin{frame}[fragile]{How to define relatedness?}
  \begin{block}{Distance}
    \begin{itemize}
      \item Euclidean distance
      \item Manhattan distance
      \item Binary distance
    \end{itemize}
  \end{block}
  \begin{block}{Similarity}
    \begin{itemize}
      \item Identity
      \item Correlation
        \vspace{.3cm}
      \item $\rightarrow$ Every similarity can be converted into distance
    \end{itemize}
  \end{block}
\end{frame}

\begin{frame}[fragile]{Euclidean distance}
  \begin{columns}
    \begin{column}{.5\linewidth}
      \begin{Definition}{Euclidean distance}
        \begin{align*}
          d(x, y) = \sqrt{\sum_i (x_i - y_i)^2}
        \end{align*}
      \end{Definition}
    \end{column}
    \begin{column}{.5\linewidth}
      \begin{center}
        \includegraphics[width=.8\linewidth]{dist_eucl.pdf}
      \end{center}
    \end{column}
  \end{columns}
\end{frame}

\begin{frame}[fragile]{Manhattan distance}
  \begin{columns}
    \begin{column}{.5\linewidth}
      \begin{Definition}{Manhattan distance}
        \begin{align*}
          d(x, y) = \sum_i |x_i - y_i|
        \end{align*}
      \end{Definition}
    \end{column}
    \begin{column}{.5\linewidth}
      \begin{center}
        \includegraphics[width=.8\linewidth]{dist_man.pdf}
      \end{center}
    \end{column}
  \end{columns}
\end{frame}

\begin{frame}[fragile]{Hierarchical clustering}
  \begin{itemize}
    \item Constructs hierarchy of clusters represented by a \alert{Cluster dendrogram}
    \item \alert{Cluster dendrogram}
      \begin{itemize}
        \item Leaf nodes: single data points
        \item Inner nodes: cluster of points
        \item Root node: all data points
      \end{itemize}
  \end{itemize}
  \begin{center}
    \includegraphics[width=.8\linewidth]{clust_dendro.png}
  \end{center}
\end{frame}

\begin{frame}[fragile]{Algorithm}
  \begin{enumerate}
    \item Start with clusters that contain only one point
    \item Compute distance between clusters
    \item Merge the two clusters with lowest distance into new cluster
    \item Go to step 2 until one single cluster remains
  \end{enumerate}
\end{frame}

\begin{frame}[fragile]{How to compute the distance between clusters?}
  \begin{center}
    \includegraphics[width=\linewidth]{clust_linkage.jpg}
  \end{center}
\end{frame}



\begin{frame}[fragile]{Data}
\begin{knitrout}\tiny
\definecolor{shadecolor}{rgb}{0.969, 0.969, 0.969}\color{fgcolor}\begin{kframe}
\begin{alltt}
\hlkwd{set.seed}\hlstd{(}\hlnum{5}\hlstd{)}
\hlstd{n} \hlkwb{=} \hlnum{5}
\hlstd{nclust} \hlkwb{=} \hlnum{3}
\hlstd{x_means} \hlkwb{=} \hlkwd{c}\hlstd{(}\hlnum{1}\hlstd{,} \hlnum{2}\hlstd{,} \hlnum{2.5}\hlstd{)}
\hlstd{y_means} \hlkwb{=} \hlkwd{c}\hlstd{(}\hlnum{1}\hlstd{,} \hlnum{2.5}\hlstd{,} \hlnum{2}\hlstd{)}
\hlstd{x} \hlkwb{=} \hlkwd{rnorm}\hlstd{(n} \hlopt{*} \hlstd{nclust,} \hlkwd{rep}\hlstd{(x_means,} \hlkwc{each}\hlstd{=n),} \hlkwc{sd}\hlstd{=}\hlnum{.4}\hlstd{)}
\hlstd{y} \hlkwb{=} \hlkwd{rnorm}\hlstd{(n} \hlopt{*} \hlstd{nclust,} \hlkwd{rep}\hlstd{(y_means,} \hlkwc{each}\hlstd{=n),} \hlkwc{sd}\hlstd{=}\hlnum{.4}\hlstd{)}
\hlstd{clust} \hlkwb{=} \hlkwd{factor}\hlstd{(}\hlkwd{rep}\hlstd{(}\hlnum{1}\hlopt{:}\hlstd{nclust,} \hlkwc{each}\hlstd{=n))}
\hlstd{id} \hlkwb{=} \hlnum{1}\hlopt{:}\hlstd{(n} \hlopt{*} \hlstd{nclust)}
\hlstd{cdata} \hlkwb{=} \hlkwd{data.frame}\hlstd{(}\hlkwc{id}\hlstd{=id,} \hlkwc{x}\hlstd{=x,} \hlkwc{y}\hlstd{=y,} \hlkwc{clust}\hlstd{=clust)}
\hlkwd{ggplot}\hlstd{(cdata,} \hlkwd{aes}\hlstd{(}\hlkwc{x}\hlstd{=x,} \hlkwc{y}\hlstd{=y))} \hlopt{+} \hlkwd{geom_point}\hlstd{(}\hlkwd{aes}\hlstd{(}\hlkwc{color}\hlstd{=clust),} \hlkwc{size}\hlstd{=}\hlnum{3}\hlstd{,} \hlkwc{show_guide}\hlstd{=F)} \hlopt{+}
  \hlkwd{geom_text}\hlstd{(}\hlkwd{aes}\hlstd{(}\hlkwc{label}\hlstd{=id,} \hlkwc{color}\hlstd{=clust),} \hlkwc{show_guide}\hlstd{=F,} \hlkwc{vjust}\hlstd{=}\hlopt{-}\hlnum{.3}\hlstd{,} \hlkwc{hjust}\hlstd{=}\hlnum{0}\hlstd{)}
\end{alltt}
\end{kframe}

{\centering \includegraphics[width=.6\linewidth]{figure/clust_data-1} 

}



\end{knitrout}
\end{frame}

\begin{frame}[fragile]{Complete linkage}
\begin{knitrout}\tiny
\definecolor{shadecolor}{rgb}{0.969, 0.969, 0.969}\color{fgcolor}\begin{kframe}
\begin{alltt}
\hlstd{cdist} \hlkwb{=} \hlkwd{dist}\hlstd{(}\hlkwd{cbind}\hlstd{(cdata}\hlopt{$}\hlstd{x, cdata}\hlopt{$}\hlstd{y),} \hlkwc{method}\hlstd{=}\hlstr{'euclidean'}\hlstd{)}
\hlstd{chclust} \hlkwb{=} \hlkwd{hclust}\hlstd{(cdist,} \hlkwc{method}\hlstd{=}\hlstr{'complete'}\hlstd{)}
\hlkwd{plot}\hlstd{(chclust,} \hlkwc{ylab}\hlstd{=}\hlstr{'Distance'}\hlstd{,} \hlkwc{xlab}\hlstd{=}\hlnum{NA}\hlstd{,} \hlkwc{main}\hlstd{=}\hlnum{NA}\hlstd{,} \hlkwc{sub}\hlstd{=}\hlnum{NA}\hlstd{)}
\end{alltt}
\end{kframe}

{\centering \includegraphics[width=.8\linewidth]{figure/clust_complete-1} 

}



\end{knitrout}
\end{frame}

\begin{frame}[fragile]{Average linkage}
\begin{knitrout}\tiny
\definecolor{shadecolor}{rgb}{0.969, 0.969, 0.969}\color{fgcolor}\begin{kframe}
\begin{alltt}
\hlstd{cdist} \hlkwb{=} \hlkwd{dist}\hlstd{(}\hlkwd{cbind}\hlstd{(cdata}\hlopt{$}\hlstd{x, cdata}\hlopt{$}\hlstd{y),} \hlkwc{method}\hlstd{=}\hlstr{'euclidean'}\hlstd{)}
\hlstd{chclust} \hlkwb{=} \hlkwd{hclust}\hlstd{(cdist,} \hlkwc{method}\hlstd{=}\hlstr{'average'}\hlstd{)}
\hlkwd{plot}\hlstd{(chclust,} \hlkwc{ylab}\hlstd{=}\hlstr{'Distance'}\hlstd{,} \hlkwc{xlab}\hlstd{=}\hlnum{NA}\hlstd{,} \hlkwc{main}\hlstd{=}\hlnum{NA}\hlstd{,} \hlkwc{sub}\hlstd{=}\hlnum{NA}\hlstd{)}
\end{alltt}
\end{kframe}

{\centering \includegraphics[width=.8\linewidth]{figure/clust_average-1} 

}



\end{knitrout}
\end{frame}

\begin{frame}[fragile]{Single linkage}
\begin{knitrout}\tiny
\definecolor{shadecolor}{rgb}{0.969, 0.969, 0.969}\color{fgcolor}\begin{kframe}
\begin{alltt}
\hlstd{cdist} \hlkwb{=} \hlkwd{dist}\hlstd{(}\hlkwd{cbind}\hlstd{(cdata}\hlopt{$}\hlstd{x, cdata}\hlopt{$}\hlstd{y),} \hlkwc{method}\hlstd{=}\hlstr{'euclidean'}\hlstd{)}
\hlstd{chclust} \hlkwb{=} \hlkwd{hclust}\hlstd{(cdist,} \hlkwc{method}\hlstd{=}\hlstr{'single'}\hlstd{)}
\hlkwd{plot}\hlstd{(chclust,} \hlkwc{ylab}\hlstd{=}\hlstr{'Distance'}\hlstd{,} \hlkwc{xlab}\hlstd{=}\hlnum{NA}\hlstd{,} \hlkwc{main}\hlstd{=}\hlnum{NA}\hlstd{,} \hlkwc{sub}\hlstd{=}\hlnum{NA}\hlstd{)}
\end{alltt}
\end{kframe}

{\centering \includegraphics[width=.8\linewidth]{figure/clust_single-1} 

}



\end{knitrout}
\end{frame}

\begin{frame}[fragile]{Data}
\begin{knitrout}\tiny
\definecolor{shadecolor}{rgb}{0.969, 0.969, 0.969}\color{fgcolor}\begin{kframe}
\begin{alltt}
\hlkwd{set.seed}\hlstd{(}\hlnum{5}\hlstd{)}
\hlstd{n} \hlkwb{=} \hlnum{5}
\hlstd{nclust} \hlkwb{=} \hlnum{3}
\hlstd{x_means} \hlkwb{=} \hlkwd{c}\hlstd{(}\hlnum{1}\hlstd{,} \hlnum{2}\hlstd{,} \hlnum{2.5}\hlstd{)}
\hlstd{y_means} \hlkwb{=} \hlkwd{c}\hlstd{(}\hlnum{1}\hlstd{,} \hlnum{2.5}\hlstd{,} \hlnum{2}\hlstd{)}
\hlstd{x} \hlkwb{=} \hlkwd{rnorm}\hlstd{(n} \hlopt{*} \hlstd{nclust,} \hlkwd{rep}\hlstd{(x_means,} \hlkwc{each}\hlstd{=n),} \hlkwc{sd}\hlstd{=}\hlnum{.4}\hlstd{)}
\hlstd{y} \hlkwb{=} \hlkwd{rnorm}\hlstd{(n} \hlopt{*} \hlstd{nclust,} \hlkwd{rep}\hlstd{(y_means,} \hlkwc{each}\hlstd{=n),} \hlkwc{sd}\hlstd{=}\hlnum{.4}\hlstd{)}
\hlstd{clust} \hlkwb{=} \hlkwd{factor}\hlstd{(}\hlkwd{rep}\hlstd{(}\hlnum{1}\hlopt{:}\hlstd{nclust,} \hlkwc{each}\hlstd{=n))}
\hlstd{id} \hlkwb{=} \hlnum{1}\hlopt{:}\hlstd{(n} \hlopt{*} \hlstd{nclust)}
\hlstd{cdata} \hlkwb{=} \hlkwd{data.frame}\hlstd{(}\hlkwc{id}\hlstd{=id,} \hlkwc{x}\hlstd{=x,} \hlkwc{y}\hlstd{=y,} \hlkwc{clust}\hlstd{=clust)}
\hlkwd{ggplot}\hlstd{(cdata,} \hlkwd{aes}\hlstd{(}\hlkwc{x}\hlstd{=x,} \hlkwc{y}\hlstd{=y))} \hlopt{+} \hlkwd{geom_point}\hlstd{(}\hlkwd{aes}\hlstd{(}\hlkwc{color}\hlstd{=clust),} \hlkwc{size}\hlstd{=}\hlnum{3}\hlstd{,} \hlkwc{show_guide}\hlstd{=F)} \hlopt{+}
  \hlkwd{geom_text}\hlstd{(}\hlkwd{aes}\hlstd{(}\hlkwc{label}\hlstd{=id,} \hlkwc{color}\hlstd{=clust),} \hlkwc{show_guide}\hlstd{=F,} \hlkwc{vjust}\hlstd{=}\hlopt{-}\hlnum{.3}\hlstd{,} \hlkwc{hjust}\hlstd{=}\hlnum{0}\hlstd{)}
\end{alltt}
\end{kframe}

{\centering \includegraphics[width=.6\linewidth]{figure/clust_data2-1} 

}



\end{knitrout}
\end{frame}

\section{PCA}
\begin{frame}
\tableofcontents[currentsection]
\end{frame}

\begin{frame}{High-dimensional data}
  \begin{itemize}
    \item \alert{Small $n$ large $p$}
    \item Few samples $n$ ($\approx 100$)
    \item Many features $p$
      \begin{itemize}
        \item $> 10000$ genes
        \item $> 27000000$ CpG sites
      \end{itemize}
  \end{itemize}
  \begin{center}
    \includegraphics[width=.5\linewidth]{microarray.jpg}
  \end{center}
\end{frame}

\begin{frame}{Problems}
  \begin{itemize}
    \item High storage costs (memory)
    \item High computational costs (time)
    \item Visualization?
    \item Curse of dimensionality
  \end{itemize}
  \begin{center}
    \includegraphics[width=.5\linewidth]{curse_dim.jpg}
  \end{center}
\end{frame}

\begin{frame}{Principle Component Analysis}
  \begin{itemize}
    \item Dimensionality reduction
    \item Visualization
    \item Missing values imputation
    \item Latent factors estimation:
      \begin{itemize}
        \item Population structure
        \item Batch-effects
        \item Cell-cycle
      \end{itemize}
  \end{itemize}
\end{frame}

\begin{frame}{Principle components}
  \begin{itemize}
    \item Minimize projection error
    \item Maximize variance
  \end{itemize}
  \begin{center}
    \includegraphics[width=.6\linewidth]{pca.jpg}
  \end{center}
\end{frame}

\begin{frame}{Pearson correlation coefficient}
  \begin{itemize}
    \item Measures \textbf{linear} dependency between $x$ and $y$
    \item $cor(x,y) = \in \left[-1,+1\right]$
    \item $cor(x,y) = 0$: no correlation
    \item $cor(x,y) = -1$: negative correlation
    \item $cor(x,y) = +1$: positive correlation
  \end{itemize}
  \begin{definition}[Pearson correlation coefficient]
    \begin{align*}
      \operatorname{cor}(x, y) = \frac{\sum_i (x_i - \bar{x})(y_i - \bar{y})}
      {\sqrt{\sum_i (x_i - \bar{x})^2}\sqrt{\sum_i (y_i - \bar{y})^2}}
    \end{align*}
  \end{definition}
\end{frame}

\begin{frame}{Pearson correlation coefficient}
  \begin{center}
    \includegraphics[width=0.8\linewidth]{pearson.png}
  \end{center}
\end{frame}

\begin{frame}{Spearman correlation coefficient}
  \begin{itemize}
    \item Measures \textbf{monotonic} dependency between $x$ and $y$
    \item Pearson correlation coefficient on rank of variables
    \item $cor(x,y) = \in \left[-1,+1\right]$
  \end{itemize}
  \begin{center}
    \includegraphics[width=0.4\linewidth]{spearman.png}
  \end{center}
\end{frame}



\begin{frame}[fragile]{Embryonic Stem Cell (ESC) data}
  \begin{itemize}
    \item Single-cell RNA-seq
    \item $n=182$ Embryonic stem cells
    \item $p=9571$ Genes
    \item Cell-cycle via Hoechst staining
  \end{itemize}
\begin{knitrout}\tiny
\definecolor{shadecolor}{rgb}{0.969, 0.969, 0.969}\color{fgcolor}

{\centering \includegraphics[width=.9\linewidth]{figure/heat_esc-1} 

}



\end{knitrout}
\end{frame}

\begin{frame}[fragile]{Correlation between two ESC}
\begin{knitrout}\tiny
\definecolor{shadecolor}{rgb}{0.969, 0.969, 0.969}\color{fgcolor}\begin{kframe}
\begin{alltt}
\hlstd{c1} \hlkwb{<-} \hlstd{esc}\hlopt{$}\hlstd{counts[}\hlkwd{which}\hlstd{(esc}\hlopt{$}\hlstd{cycle} \hlopt{==} \hlstr{'G1'}\hlstd{)[}\hlnum{1}\hlstd{],]}
\hlstd{c2} \hlkwb{<-} \hlstd{esc}\hlopt{$}\hlstd{counts[}\hlkwd{which}\hlstd{(esc}\hlopt{$}\hlstd{cycle} \hlopt{==} \hlstr{'S'}\hlstd{)[}\hlnum{1}\hlstd{],]}
\end{alltt}
\end{kframe}
\end{knitrout}
  \begin{columns}[c]
    \begin{column}{.5\linewidth}
\begin{knitrout}\tiny
\definecolor{shadecolor}{rgb}{0.969, 0.969, 0.969}\color{fgcolor}\begin{kframe}
\begin{alltt}
\hlkwd{qplot}\hlstd{(c1, c2,} \hlkwc{xlab}\hlstd{=}\hlstr{'G1 cell'}\hlstd{,} \hlkwc{ylab}\hlstd{=}\hlstr{'S cell'}\hlstd{)}
\end{alltt}
\end{kframe}

{\centering \includegraphics[width=\linewidth]{figure/cor_c1_c2-1} 

}



\end{knitrout}
    \end{column}
    \begin{column}{.5\linewidth}
\begin{knitrout}\tiny
\definecolor{shadecolor}{rgb}{0.969, 0.969, 0.969}\color{fgcolor}\begin{kframe}
\begin{alltt}
\hlkwd{cor}\hlstd{(c1, c2,}
    \hlkwc{method}\hlstd{=}\hlstr{'pearson'}\hlstd{)}
\end{alltt}
\begin{verbatim}
## [1] 0.547453
\end{verbatim}
\begin{alltt}
\hlkwd{cor}\hlstd{(c1, c2,}
    \hlkwc{method}\hlstd{=}\hlstr{'spearman'}\hlstd{)}
\end{alltt}
\begin{verbatim}
## [1] 0.5785747
\end{verbatim}
\end{kframe}
\end{knitrout}
    \end{column}
  \end{columns}
\end{frame}

\begin{frame}[fragile]{Correlation matrix}
\begin{knitrout}\tiny
\definecolor{shadecolor}{rgb}{0.969, 0.969, 0.969}\color{fgcolor}\begin{kframe}
\begin{alltt}
\hlstd{cor_cells} \hlkwb{<-} \hlkwd{cor}\hlstd{(}\hlkwd{t}\hlstd{(esc}\hlopt{$}\hlstd{counts),} \hlkwc{method}\hlstd{=}\hlstr{'pearson'}\hlstd{)}
\hlkwd{heatmap}\hlstd{(cor_cells,} \hlkwc{Rowv}\hlstd{=}\hlnum{NA}\hlstd{,} \hlkwc{Colv}\hlstd{=}\hlnum{NA}\hlstd{)}
\end{alltt}
\end{kframe}

{\centering \includegraphics[width=.5\linewidth]{figure/cor_matrix-1} 

}



\end{knitrout}
\end{frame}

\begin{frame}[fragile]{PCA}
\begin{knitrout}\tiny
\definecolor{shadecolor}{rgb}{0.969, 0.969, 0.969}\color{fgcolor}\begin{kframe}
\begin{alltt}
\hlstd{svd} \hlkwb{<-} \hlkwd{svd}\hlstd{(}\hlkwd{scale}\hlstd{(esc}\hlopt{$}\hlstd{counts))}
\hlstd{dsvd} \hlkwb{<-} \hlkwd{as.data.frame}\hlstd{(svd}\hlopt{$}\hlstd{u[,} \hlnum{1}\hlopt{:}\hlnum{10}\hlstd{])}
\hlkwd{colnames}\hlstd{(dsvd)} \hlkwb{<-} \hlkwd{paste0}\hlstd{(}\hlstr{'pc'}\hlstd{,} \hlkwd{seq}\hlstd{(}\hlnum{1}\hlstd{,} \hlkwd{ncol}\hlstd{(dsvd)))}
\hlstd{dsvd}\hlopt{$}\hlstd{cycle} \hlkwb{<-} \hlstd{esc}\hlopt{$}\hlstd{cycle}
\hlstd{ve} \hlkwb{<-} \hlstd{svd}\hlopt{$}\hlstd{d}\hlopt{^}\hlnum{2} \hlopt{/} \hlkwd{sum}\hlstd{(svd}\hlopt{$}\hlstd{d}\hlopt{^}\hlnum{2}\hlstd{)}
\hlkwd{barplot}\hlstd{(ve[}\hlnum{1}\hlopt{:}\hlnum{10}\hlstd{],} \hlkwc{xlab}\hlstd{=}\hlstr{'Principle components'}\hlstd{,} \hlkwc{ylab}\hlstd{=}\hlstr{'% Variance explained'}\hlstd{,}
        \hlkwc{col}\hlstd{=}\hlstr{'lightblue'}\hlstd{)}
\end{alltt}
\end{kframe}

{\centering \includegraphics[width=.8\linewidth]{figure/pca_ve-1} 

}



\end{knitrout}
\end{frame}

\begin{frame}[fragile]{PC1 versus PC2}
\begin{knitrout}\tiny
\definecolor{shadecolor}{rgb}{0.969, 0.969, 0.969}\color{fgcolor}\begin{kframe}
\begin{alltt}
\hlkwd{ggplot}\hlstd{(dsvd,} \hlkwd{aes}\hlstd{(}\hlkwc{x}\hlstd{=pc1,} \hlkwc{y}\hlstd{=pc2,} \hlkwc{color}\hlstd{=cycle))} \hlopt{+} \hlkwd{geom_point}\hlstd{(}\hlkwc{size}\hlstd{=}\hlnum{3}\hlstd{)}
\end{alltt}
\end{kframe}

{\centering \includegraphics[width=.7\linewidth]{figure/pca_12-1} 

}



\end{knitrout}
\end{frame}

\begin{frame}[fragile]{Correlation PC1 and cell-cycle}
\begin{knitrout}\tiny
\definecolor{shadecolor}{rgb}{0.969, 0.969, 0.969}\color{fgcolor}\begin{kframe}
\begin{alltt}
\hlkwd{ggplot}\hlstd{(dsvd,} \hlkwd{aes}\hlstd{(}\hlkwc{x}\hlstd{=cycle,} \hlkwc{y}\hlstd{=pc1,} \hlkwc{color}\hlstd{=cycle))} \hlopt{+} \hlkwd{geom_boxplot}\hlstd{()} \hlopt{+} \hlkwd{geom_jitter}\hlstd{(}\hlkwc{position}\hlstd{=}\hlkwd{position_jitter}\hlstd{(}\hlkwc{width}\hlstd{=}\hlnum{.1}\hlstd{))}
\end{alltt}
\end{kframe}

{\centering \includegraphics[width=.7\linewidth]{figure/pca_cycle1-1} 

}



\end{knitrout}
\end{frame}

\begin{frame}[fragile]{Accounting for cell-cycle}
\begin{knitrout}\tiny
\definecolor{shadecolor}{rgb}{0.969, 0.969, 0.969}\color{fgcolor}\begin{kframe}
\begin{alltt}
\hlstd{svd_compress} \hlkwb{<-} \hlkwa{function}\hlstd{(}\hlkwc{svd}\hlstd{,} \hlkwc{k}\hlstd{=}\hlnum{1}\hlstd{) \{}
  \hlkwd{return} \hlstd{(svd}\hlopt{$}\hlstd{u[,} \hlnum{1}\hlopt{:}\hlstd{k,} \hlkwc{drop}\hlstd{=}\hlnum{FALSE}\hlstd{]} \hlopt{%*%} \hlkwd{diag}\hlstd{(svd}\hlopt{$}\hlstd{d[}\hlnum{1}\hlopt{:}\hlstd{k],} \hlkwc{nrow}\hlstd{=k)} \hlopt{%*%} \hlkwd{t}\hlstd{(svd}\hlopt{$}\hlstd{v)[}\hlnum{1}\hlopt{:}\hlstd{k,,} \hlkwc{drop}\hlstd{=}\hlnum{FALSE}\hlstd{])}
\hlstd{\}}
\hlstd{counts_pc1} \hlkwb{<-} \hlkwd{svd_compress}\hlstd{(svd,} \hlnum{1}\hlstd{)}
\hlstd{counts_npc1} \hlkwb{<-} \hlstd{esc}\hlopt{$}\hlstd{counts} \hlopt{-} \hlstd{counts_pc1}
\end{alltt}
\end{kframe}
\end{knitrout}
\end{frame}

\begin{frame}[fragile]{Raw counts}
\begin{knitrout}\tiny
\definecolor{shadecolor}{rgb}{0.969, 0.969, 0.969}\color{fgcolor}

{\centering \includegraphics[width=\linewidth]{figure/heat_esc2-1} 

}



\end{knitrout}
\end{frame}

\begin{frame}[fragile]{Counts explained by cell-cycle}
\begin{knitrout}\tiny
\definecolor{shadecolor}{rgb}{0.969, 0.969, 0.969}\color{fgcolor}

{\centering \includegraphics[width=\linewidth]{figure/heat_esc_pc1-1} 

}



\end{knitrout}
\end{frame}

\begin{frame}[fragile]{Counts after adjusting for cell-cycle}
\begin{knitrout}\tiny
\definecolor{shadecolor}{rgb}{0.969, 0.969, 0.969}\color{fgcolor}

{\centering \includegraphics[width=.9\linewidth]{figure/heat_esc_npc1-1} 

}



\end{knitrout}
\end{frame}


\include{appendix}


\end{document}
